lib/ur/failurereporting.go

```diff
@@ -65,37 +67,20 @@ type failureStat struct {
 }
 
 func (h *failureHandler) Serve(ctx context.Context) error {
-	go func() {
-		select {
-		case h.optsChan <- h.cfg.Options():
-		case <-ctx.Done():
-		}
-	}()
 	h.cfg.Subscribe(h)
 	defer h.cfg.Unsubscribe(h)
 
-	var url string
+	url, sub, evChan := h.applyOpts(h.cfg.Options(), nil)
+
+	h.handleOldFailureReports(ctx, url, sub != nil)
```

Comment:
This is really a PR of its own hidden in here

Reply:
https://github.com/syncthing/syncthing/pull/7164